article/background.tex

```diff
@@ -4,107 +4,69 @@ \section{Background}
+
+\subsection{Software Dependency Problem and Containers}
+
+A fundamental challenge of software dependency management is that dependencies can be nested arbitrarily deeply, which increases in complexity (TODO: O^n? does that even make sense to say?)
+As soon as one version of one dependency eventually becomes unavailable, the first step of executability is manually solving a computationally difficult problem.
+One way of solving the issue of rebuilding environments with integrated dependencies is distributing an entire operating system image with the environment installed and configured, which can be emulated as a "guest" environment within the "host" machine.
+Virtual machines (colloquially "VMs") are attractive solutions to the software dependency problem, because the artifact that is distributed is in working order and should be executable with almost no knowledge of the internal workings.
+Once running, guest machines are self-contained and isolated from the host, which then eliminates the posibility of polluting the host environment and allows system administrators to safely allow relatively unrestricted usage from semi-trusted users.
+The weakness of VMs for solving this problem is primarily that they are expensive; VMs are inefficient with resource allocation in general. \cite{TODO: asmacdo} \footnote{Hard drive usage(and bandwidth), memory allocation, CPU allocation, start/restart time}
+
+Modern advances in "container technology" have allowed similar benefits but strip redundancy by making limited use of the host machine, specifically the hypervisor.
+Containers enable a complete working environment often as small as a few Megabytes, and can it be started as quickly as a normal local process.
+Over time Docker and other organizations have established a governing body under the Linux Foundation project, the "Open Container Initiative" (OCI).
+OCI produced an specification for containers, which can be used by various container runtimes and toolsets including Docker, Podman, and Apptainer \cite{TODO: apptainer something}
+While OCI images are nearly ubiquitous in the cloud industry, Singularity (recently renamed to Apptainer) is a toolset that was developed for High Performance Computing, and has its own native specification.
+
+\subsection{Preserving Version Collection of Modular Components}
+
+TODO: weak needs a lot more
+A proposed set of principles, "Yodas Organigram for Data Analysis" (YODA Principles), suggests linking external repositories via git submodules. \cite{TODO YODA}
+By distributing a set of separately versioned component parts, YODA "compliant" data analysis helps us to make use of modularity to improve portability, reusability, and maintainability of the research execution stack.
+the basic feasibility of \textit{de novo} research output generation from the earliest recorded provenance
+TODO: needs a little datalad but not too much
+
+
+\subsection{Target Article 2022, Original Execution}
```

Comment:
Few people (14 citations) care about this article in particular. Lots of people care about the general case of fMRI analysis, which this article happens to exemplify with reexecutable code. Better describe the data analysis and mention in synchronicity that that is exactly what the article does.